OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -236,5 +274,27 @@ void Explode(World world)
 
 			args.Weapon.Impact(Target.FromPos(pos), args.SourceActor, args.DamageModifiers);
 		}
+
+		bool AnyValidTargetsInRadius(World world, WPos pos, WDist radius, Actor firedBy, bool checkTargetType)
```

Comment:
Seems reasonable, but pretty big for this PR. It would probably be better to take this PR with the duplication and then fix this in a followup.